packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1,3 +1,55 @@
+//
+// XXX_vincent We should restructure outputs.settingsByRoomId to be a DefaultMap of
+// a WrappedSinglePageResource. The WrappedSinglePageResource is both a derived
+// signal, and a way to waitUntilLoaded().
+//
+// So it is a lot like our existing SinglePageResource class, except that its
+// exposed value ultimately is a derived signal to obtain the AsyncResult.
+//
+// ------------------------------------------------------------------------
+//
+// This way, we would no longer need to do this:
+//
+//   const settings = useSyncExternalStoreWithSelector(
+//     store.outputs.settingsByRoomId.subscribe,
+//     () => store.getNotificationSettingsLoadingState(room.id),
+//     () => store.getNotificationSettingsLoadingState(room.id),
+//     identity,
+//     shallow2
+//   );
+//
+// But we could do this instead:
+//
+//   const settings = useSignal(
+//     store.outputs.settingsByRoomId.getOrCreate(room.id).signal,
+//   );
+//
+// ------------------------------------------------------------------------
+//
+// And similarly:
+//
+//   useEffect(
+//     () => void store.waitUntilRoomNotificationSettingsLoaded(room.id)
+//   )
+//
+// Could become:
+//
+//   useEffect(() =>
+//     store.outputs.settingsByRoomId.getOrCreate(room.id).waitUntilLoaded()
+//   )
+//
+// It would allow us to remove both of these methods:
+// - `getNotificationSettingsLoadingState()`
+// - `waitUntilRoomNotificationSettingsLoaded()`
+//
+// ------------------------------------------------------------------------
+//
+// These signals also whenever one of their sources change:
+//
+// 1. The loading state changes (from a fetch).
+// 2. The base settings themselves are updated (side-effect of a fetch).
+// 3. Any optimistic updates are applied.
+//
```

Comment:
This idea is now implemented (in 59ab0b44c0391ffb8069c95c3ed0cccbf83209c6). I've left this comment in, because I think room/user threads can follow suit in similar fashion.